### DISCUSSION TITLE
Problems combining loom file with my adata object

### DISCUSSION THREAD

**Discussion by Pawan291** [theislab/scvelo#1006] [General]
Hello everyone, 
I am quite new to scvelo package and just trying to combine my processed adata object from a single sample to a loom file. Here is the code and the output.
Code:
adata = scv.read(folder + "UMAP regressed.h5ad", cache=True)
ldata = scv.read(folder + "loomDWSoloRNA.loom", cache=True)
adata = scv.utils.merge(adata, ldata)

Output for adata before merging with loom file:
AnnData object with n_obs × n_vars = 4404 × 21445
    obs: 'identifier', 'name', 'cells', 'grouping', 'n_counts', 'n_genes', 'percent.mito', 'S_score', 'G2M_score', 'phase', 'louvain'
    var: 'n_cells', 'means', 'dispersions', 'dispersions_norm', 'highly_variable', 'mean', 'std'
    uns: 'grouping_colors', 'louvain', 'louvain_colors', 'name_colors', 'neighbors', 'pca', 'umap'
    obsm: 'X_pca', 'X_umap'
    varm: 'PCs'
    layers: 'counts'
    obsp: 'connectivities', 'distances'

Output after merging with loom file:
AnnData object with n_obs × n_vars = 4404 × 156
    obs: 'identifier', 'name', 'cells', 'grouping', 'n_counts', 'n_genes', 'percent.mito', 'S_score', 'G2M_score', 'phase', 'louvain', 'nCount_RNA', 'nFeature_RNA', 'orig.ident'
    var: 'n_cells', 'means', 'dispersions', 'dispersions_norm', 'highly_variable', 'mean', 'std'
    uns: 'grouping_colors', 'louvain', 'louvain_colors', 'name_colors', 'neighbors', 'pca', 'umap'
    obsm: 'X_pca', 'X_umap'
    varm: 'PCs'
    layers: 'counts'
    obsp: 'connectivities', 'distances'

As you can see the number of cells remains the same but the number of genes reduced from 21445 to 156. I don't really understand the problem here and I am using scvelo tutorial for the setup of the pipeline for my datasets. 

Thank you very much for the help and I wish you a good day.

Pawandeep

**Comment by lwf20** [comment]
how many genes are in your `ldata` object? I met a similar problem before and I finally find my `ldata.var_names` is not the same as `adata.var_name`.

**Reply by Pawan291** [reply]
Hey, thanks for the reply. I don't find any ldata.var_names in the ldata object. Infact there is no var_names in ldata object.

**Reply by lwf20** [reply]
Hi, I check my code and there is ldata.var_names in my dataset. I think some BUG happened when you create your loom files

![图片](https://user-images.githubusercontent.com/81340626/220804152-04ae4920-9cbe-4a98-aca0-c3bdd8091bca.png)
